### ISSUE TITLE
Incorrect behavior when transferring the type in IntEnum

### ISSUE THREAD

**Issue by gitlab-importer** [pypy/pypy#3912]
In Heptapod by @xxm on Apr 19, 2023, 02:48

The following code defines a new class called EnumTest which is a subclass of IntEnum. The EnumTest class has two members: A with a value of 0 and C with a value of 1. The __str__ method is overridden for the EnumTest class to return the lowercased name of the member when an instance of the class is converted to a string.

The code then creates an instance of the EnumTest class called s with a value of EnumTest.A. Finally, the code prints the hexadecimal representation of s using the :X format specifier and the lowercase name of s using the str() function.

So the expected output of this code will be: 0 a. The first value is the hexadecimal representation of the s instance, which is 0 because the value of EnumTest.A is 0. The second value is the lowercase name of the s instance, which is "a" because the name of EnumTest.A is "A" and the __str__ method was defined to return the lowercase name. However, Pypy reports a ValueError.

By the way, it seems that the early versions of CPython, e.g., 3.9.0 reports the same error as Pypy. This case can work as expected on the latest CPython.  

test.py
```
from enum import IntEnum

class EnumTest(IntEnum):
     A = 0
     C = 1
    
     def __str__(self):
             return self.name.lower()

s = EnumTest.A
print(f"{s:X} {str(s)}")
```

Expected output (Behaviors on CPython 3.11.3): 0 a

Actual output in PyPy3.9:
```
Traceback (most recent call last):
  File "/home/xxm/Desktop/pypy/test.py", line 1360, in <module>
    print(f"{s:X} {str(s)}")
  File "/home/xxm/Desktop/pypy/pypy3.9-v7.3.11-linux64/pypy3.9-v7.3.11-linux64/lib/pypy3.9/enum.py", line 778, in __format__
    return cls.__format__(val, format_spec)
ValueError: Unknown format code 'X' for object of type 'str'
```

Reproduce step:
1. download the binaries from https://www.pypy.org/download.html
2. 'pypy3.9-v7.3.11-linux64/bin/pypy3.9' test.py

Environment:
Ubuntu 18.04
PyPy:pypy3.9-v7.3.11-linux64
CPython: 3.11.3

**Comment by mattip** [comment]
Reopening. There is some problem in `_operator.index`:
```
$cat /tmp/test_enum.py
import enum
import _operator

class A(int, enum.Enum):
    TEST=1

print(type(_operator.index(A.TEST)))

$ cpython3.10 /tmp/test_enum.py 
<class 'int'>
$ cpython3.11 /tmp/test_enum.py 
<class 'int'>
$ pypy3.10/bin/pypy /tmp/test_enum.py
<enum 'A'>
$ pypy3.11/bin/pypy /tmp/test_enum.py
<enum 'A'>
```

**Comment by mattip** [comment]
Found it here on line 575, the condition should be exactly space.w_int, not a subclass https://github.com/pypy/pypy/blob/3e8d49a3c060804faed23d8361cef40a3ee7c41d/pypy/objspace/descroperation.py#L571-L576

Fixed in 05cf640402c
